### PR TITLE
Merge /etc before dropping the snapshot

### PR DESCRIPTION
**Pull request by aplanas**

### CODE REVIEW

**Comment by aplanas** [comment]
Do not merge yet, some trivial changes in /etc generate invalid data: `transactional-update --drop-if-no-change run touch /etc/test` is generating broken files in my setup

**Comment by thkukuk** [comment]
On Mon, Sep 07, Alberto Planas wrote:

>   • Mount /run in the chroot

That's a really bad idea, as this can have a high impact on the
running system, thus the update is no longer atomic.

**Comment by aplanas** [comment]
Drop `Mount /run in the chroot`. If /run is mounted, will be only under the `run` sub-command and not in the general case.

File: sbin/transactional-update.in [hunk 1]
```diff
@@ -290,9 +292,17 @@ teardown() {
...
 	INOTIFY_PID=""
+	# Request the merge of the new overlay if present
+	ETC_OVERLAY_MERGE=1
 	quit 0
     fi
 
+    # If needed, merge the orverlay from the snapshot to the current /etc
+    if [ ! -z "${ETC_OVERLAY_MERGE}" ] ; then
+	log_error "Merging overlay directory ${SNAPSHOT_DIR}/etc into /etc..."
+	rsync --archive --inplace --xattrs --filter='-x security.selinux' --acls --exclude 'fstab' --delete --quiet "${SNAPSHOT_DIR}/etc/" /etc
```
Comment (laenion) [inline sbin/transactional-update.in:303]: Why don't you move this block up into the rollback section 3 lines above? I don't see any immediate reason to use an additional variable / separate step for this...

**Review by laenion — CHANGES_REQUESTED** [review]

File: sbin/transactional-update.in [hunk 2]
```diff
@@ -82,7 +82,9 @@ MOUNT_DIR=""
 SNAPPER_NO_DBUS=""
 TMPFILE=""
 ETC_OVERLAY_DIR=""
+ETC_OVERLAY_DIR_PREV=""
```
[the remark this answers is not in the document]
Comment (aplanas) [reply sbin/transactional-update.in:85]: https://github.com/openSUSE/transactional-update/blob/master/sbin/transactional-update.in#L990 uses it. Do you want me to remove it?

File: sbin/transactional-update.in [hunk 3]
```diff
@@ -290,9 +292,17 @@ teardown() {
 	kill ${INOTIFY_PID}
 	log_info "The snapshot does not contain any changed files."
 	INOTIFY_PID=""
+	# Request the merge of the new overlay if present
+	ETC_OVERLAY_MERGE=1
 	quit 0
     fi
 
+    # If needed, merge the orverlay from the snapshot to the current /etc
```
[the remark this answers is not in the document]
Comment (aplanas) [reply sbin/transactional-update.in:300]: Yes, done.

File: sbin/transactional-update.in (hunk 1 above)
Comment (aplanas) [reply sbin/transactional-update.in:303]: Indeed.

**Comment by laenion** [comment]
This change is necessary for Salt integration as discussed in https://etherpad.suse.de/p/salt-microos.

**Pull request merged**